google-cloud-spanner/src/test/java/com/google/cloud/spanner/DatabaseClientImplTest.java

```diff
@@ -195,6 +198,160 @@ public void tearDown() {
     mockSpanner.removeAllExecutionTimes();
   }
 
+  @Test
+  public void testPoolMaintainer_whenInactiveTransactions_removeSessionsFromPool() {
+    InactiveTransactionRemovalOptions inactiveTransactionRemovalOptions =
+        InactiveTransactionRemovalOptions.newBuilder()
+            .setExecutionTimeThreshold(
+                Duration.ofMillis(5L)) // anything more than 5s will be long-running
+            .setRecurrenceDuration(Duration.ofSeconds(15L))
+            .build();
+    SessionPoolOptions sessionPoolOptions =
+        SessionPoolOptions.newBuilder()
+            .setMinSessions(1)
+            .setMaxSessions(1) // to ensure there is 1 session and pool is 100% utilized
+            .setCloseIfInactiveTransactions()
+            .setInactiveTransactionRemovalOptions(inactiveTransactionRemovalOptions)
+            .build();
+    spanner =
+        SpannerOptions.newBuilder()
+            .setProjectId(TEST_PROJECT)
+            .setDatabaseRole(TEST_DATABASE_ROLE)
+            .setChannelProvider(channelProvider)
+            .setCredentials(NoCredentials.getInstance())
+            .setSessionPoolOption(sessionPoolOptions)
+            .build()
+            .getService();
+    DatabaseClientImpl client =
+        (DatabaseClientImpl)
+            spanner.getDatabaseClient(DatabaseId.of(TEST_PROJECT, TEST_INSTANCE, TEST_DATABASE));
+    Instant initialExecutionTime = client.pool.poolMaintainer.lastExecutionTime;
+
+    try (TransactionManager manager = client.transactionManager()) {
+      TransactionContext transaction = manager.begin();
+      while (true) {
+        try {
+          // Simulate a delay of 20s to ensure that the below transaction is a long-running one.
+          // We require to wait for 20s so that the main thread executes at-least once every 10s
+          // As per this test, anything which takes more than 5s is long-running
+          mockSpanner.setExecuteSqlExecutionTime(
+              SimulatedExecutionTime.ofMinimumAndRandomTime(
+                  (int) Duration.ofSeconds(20).toMillis(), 0));
```

Comment:
Yes, 20s is a large value. The reason I couldn't tweak it to be lower is since the main thread (for session maintenance) executes every 10s. I'll see if I can configure this to be a lower value, say every 2s and probably tweak all the `InactiveTransactionRemovalOptions` in order of milli-seconds and not seconds.